skip_unique_checks storage param (#23214)
cockroachdb/docs: src/current/v26.2/alter-index.md

For usage, see [Synopsis](#synopsis).

### `SET`

`ALTER INDEX ... SET (storage_parameter)` modifies an alterable [index storage parameter]({% link {{ page.version.version }}/with-storage-parameter.md %}#index-parameters) on an existing index.

{{site.data.alerts.callout_info}}
Most [index storage parameters]({% link {{ page.version.version }}/with-storage-parameter.md %}#index-parameters) can only be set at index creation time and cannot be modified afterward.

> {{site.data.alerts.end}}

Reviewer: Suggest removing this callout. It feels like overkill given that the definition already suggests that not every parameter is "alterable", and the target header on this same page already has that info.

The table is still split into ranges at `25.00`, `50.00`, and `75.00`, but the `split_enforced_until` column is now `NULL` for all ranges in the table. The split is no longer enforced, and CockroachDB can [merge the data]({% link {{ page.version.version }}/architecture/distribution-layer.md %}#range-merges) in the table as needed.

### Modify index storage parameters

> Most [index storage parameters]({% link {{ page.version.version }}/with-storage-parameter.md %}#index-parameters) can only be set when creating an index and cannot be changed afterward. However, some index storage parameters can be modified on an existing index using `ALTER INDEX ... SET` or `ALTER INDEX ... RESET`.

Reviewer: ```suggestion
Many [index storage parameters]({% link {{ page.version.version }}/with-storage-parameter.md %}#index-parameters) can only be set when creating an index and cannot be changed afterward. However, some index storage parameters can be modified on an existing index using `ALTER INDEX ... SET` or `ALTER INDEX ... RESET`.
```

Reviewer: "Many" feels more future-proof

> This example demonstrates enabling and disabling the `skip_unique_checks` parameter on a unique index on a [`REGIONAL BY ROW`]({% link {{ page.version.version }}/regional-tables.md %}) table.

Reviewer: ```suggestion
This example demonstrates modifying the `skip_unique_checks` parameter on a unique index on a [`REGIONAL BY ROW`]({% link {{ page.version.version }}/regional-tables.md %}) table.
```

Reviewer: Less wordy and matches the existing nomenclature on the page

{% include_cached copy-clipboard.html %}
~~~ sql
CREATE DATABASE multi_region_db PRIMARY REGION "us-east1" REGIONS "us-west1", "europe-west1";
USE multi_region_db;

CREATE TABLE users (
  id UUID PRIMARY KEY DEFAULT gen_random_uuid(),
  email STRING UNIQUE,
  name STRING
) LOCALITY REGIONAL BY ROW;
~~~

View the index on the `email` column:

{% include_cached copy-clipboard.html %}
~~~ sql
SHOW CREATE TABLE users;
~~~

~~~
  table_name |                                         create_statement
-------------+----------------------------------------------------------------------------------------------------
  users      | CREATE TABLE public.users (
             |     id UUID NOT NULL DEFAULT gen_random_uuid(),
             |     email STRING NULL,
             |     name STRING NULL,
             |     crdb_region public.crdb_internal_region NOT VISIBLE NOT NULL DEFAULT default_to_database_primary_region(gateway_region())::public.crdb_internal_region,
             |     CONSTRAINT users_pkey PRIMARY KEY (id ASC),
             |     UNIQUE INDEX users_email_key (email ASC)
             | ) LOCALITY REGIONAL BY ROW
(1 row)
~~~

> The `users_email_key` index is a unique index that is implicitly partitioned by the `crdb_region` column. By default, CockroachDB performs cross-partition uniqueness checks for this index (`skip_unique_checks = false`).

Reviewer: ```suggestion
The `users_email_key` index is a unique index that is implicitly partitioned by the `crdb_region` column. By default, CockroachDB performs cross-partition uniqueness checks for this index because `skip_unique_checks` is set to `false`.
```

Reviewer: Making it more clear what the setting definition has to do with the statement. The second usage under **Reset a storage parameter** is fine because this modification introduces the relationship
